eng/pipelines/templates/Build_and_UnitTest.yml

```diff
@@ -234,7 +234,7 @@ steps:
     solution: "build\\build.proj"
     msbuildVersion: "16.0"
     configuration: "$(BuildConfiguration)"
-    msbuildArguments: "/t:Pack /p:BuildRTM=$(BuildRTM) /p:ExcludeTestProjects=$(BuildRTM) /p:BuildNumber=$(BuildNumber)"
+    msbuildArguments: "/t:Pack /p:BuildRTM=$(BuildRTM) /p:ExcludeTestProjects=$(BuildRTM) /p:BuildNumber=$(BuildNumber) /p:ImportNuGetBuildTasksPackTargetsFromSdk=true /p:NuGetPackTaskAssemblyFile=$(Build.Repository.LocalPath)\\artifacts\\NuGet.Build.Tasks.Pack\\16.0-RTM\\bin\\Release\\net472\\NuGet.Build.Tasks.Pack.dll /p:NuGetBuildTasksPackTargets=$(Build.Repository.LocalPath)\\artifacts\\NuGet.Build.Tasks.Pack\\16.0-RTM\\bin\\Release\\net472\\NuGet.Build.Tasks.Pack.targets"
```

Comment:
Should the local builds do the same thing? 

I think we should try to avoid discrepancies between CI & local builds. 

Some of the path values can/should be variables as well.

Reply:
Additionally, this would create some churn when we add new features. 

Take licenses, readme and icons for example. We'd have to suppress so many warnings because the server usually isn't ready by the time the client is.

Reply:
This change is one of the outcomes from https://github.com/NuGet/Client.Engineering/issues/726, I believe (/cc @zivkan).

> Should the local builds do the same thing?
>
> I think we should try to avoid discrepancies between CI & local builds.

I do think it's a good idea to do this, in general, including for local builds.

> Some of the path values can/should be variables as well.

I agree. I couldn't really figure out where those would belong and I was hoping someone would just go "oh yeah just use this var over here", cause I couldn't find anything.

> Take licenses, readme and icons for example. We'd have to suppress so many warnings because the server usually isn't ready by the time the client is.

I don't believe any of these would have caused any churn, because they're all opt-in features. They would've just been extra files, as far as the server is concerned.

Reply:
> I do think it's a good idea to do this, in general, including for local builds.

I think the PR should handle both then. 

> I agree. I couldn't really figure out where those would belong and I was hoping someone would just go "oh yeah just use this var over here", cause I couldn't find anything.

Look at the build variables in the build definition. When you edit it, they should appear. Someone from the CI initiative might have a better idea, but I think it's largely hunting how the output path is generated.

> I don't believe any of these would have caused any churn, because they're all opt-in features. They would've just been extra files, as far as the server is concerned.

We use warnings as errors, so the moment the warnings for licenses/icons appear, this would fail the build. 

I guess I just think using NuGet.Client repo as another test is a bit of an overkill. :)

Reply:
> I'm just not confident we're getting much

If the cost to implement is similarly low, it can still be worthwhile.

> Take licenses, readme and icons for example. We'd have to suppress so many warnings because the server usually isn't ready by the time the client is.

That's a feature, not a bug. Customers were impacted by the fact that we started warning about iconUrl being deprecated before nuget.org even supported embedded icons.  If we saw that our own builds are affected, then hopefully the engineer who created the PR would realize this is going to have a bad customer impact and go back to the design to allow both iconUrl and embedded icons to co-exist for some period of time and re-introduce the warning in a later release. This would have been a customer experience win.

> I guess I just think using NuGet.Client repo as another test is a bit of an overkill

If it has low cost/technical debt, and it can help us find bugs (especially if our automated tests don't catch it), then it has a customer win.

As Kat pointed out, this was in response to an incident where we failed to create some packages, but the build didn't fail, so we didn't notice, and after publishing the incomplete set of packages, customers and tools that quickly/automatically updated were broken. While we did other thing to try to prevent this incident happening again, it shows there are gaps in our tests. Hence, if it's easy/cheap enough, I see it as worthwhile.

Reply:
> If the cost to implement is similarly low, it can still be worthwhile.

I think CI & local build should be equivalent whenever possible. 
Having discrepancies could cause randomness.

> That's a feature, not a bug. 

Not my point though. The rest of the your comment also holds no value. We don't do anything special in pack, most of us don't run changed about our packed projects.

> If it has low cost/technical debt, and it can help us find bugs (especially if our automated tests don't catch it), then it has a customer win.

That's the part I disagree with. This is not a test. 

There are other build tools that self-test, but their tests are more nuanced. For example, roslyn does 2x builds. 

1. Build the compiler with a pre-built roslyn.
1. Build the exact same code, with the new compiler

and then compare the 2 outputs. 

We're not doing that here. It's a `test` without an assertion.

I don't want to block this PR, I just this is a pretty much a net 0 change.

Reply:
> I think CI & local build should be equivalent whenever possible.
I've added this to the local build as well, btw.

> There are other build tools that self-test, but their tests are more nuanced. For example, roslyn does 2x builds.

What if we only do this for NonRTM (enabling pack for NonRTM in the process), instead of only for RTM?

Reply:
> What if we only do this for NonRTM (enabling pack for NonRTM in the process), instead of only for RTM?

I don't think that addresses my concern. How would we know something changed?

Reply:
this wouldn't be to compare outputs, per se. It would be to have a non-critical path be the one that blows up in case of unexpected warnings.

Reply:
> blows up in case of unexpected warnings.

We have actual tests for that. 

Anyway, I think if *any* scenario uses built pack, then *all* of them should. 

Changing the build script doesn't guarantee that. Bootstrapping through the props/targets ensures things are always consistent.

Reply:
> Bootstrapping through the props/targets ensures things are always consistent.

So if I added this sort of thing over at `common.targets`/`common.project.props`, you'd be ok with this change?

Reply:
I still don't think there's much benefit to this change, but as long as the dev experience is consistent with the CI and all relevant scripts, I won't argue against it.